### PR TITLE
chore: [CAR-6012] Updated typescript-eslint/parser version

### PR DESCRIPTION
**Pull request by umrzoq-toshkentov**
**What's Done**
Updated typescript-eslint/parser version to 6.21.0

**Issue Link**
https://superdispatch.atlassian.net/browse/CAR-6012

### CODE REVIEW

**Review by SiriusCrain — APPROVED** [review]

File: packages/eslint-plugin/package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
   },
   "dependencies": {
     "@typescript-eslint/eslint-plugin": "^5.8.0",
```
Comment (doniyor2109) [inline packages/eslint-plugin/package.json:18]: Can we upgrade all @typescript-eslint scoped packages?

Comment (umrzoq-toshkentov) [reply packages/eslint-plugin/package.json:18]: okay let's try it

Comment (umrzoq-toshkentov) [reply packages/eslint-plugin/package.json]: updated

**Comment by umrzoq-toshkentov** [comment]
@doniyor2109 sure, can you help to merge this PR please?

**Pull request merged**